files/demo.hledger

```diff
@@ -0,0 +1,178 @@
+# about that because you will make use of implicit posting amount discussed in previous examples but anyway it is important to understand. By the other hand, Expenses are positive,
+# the opposite of what we may expect because it will decrease the balance in our Asset Account. Think of it like a Revenue is negative for the other party and an Expense is positive
+# for him. So don't worry when you see Revenue as negative in your reports, that's the expected behavior.
+
+* Now it is your turn. Create 2 expenses and think about what is the best account other than the bank with implicit amount that help you to organize you finances
+* After you are done, go to the terminal, in the folder where this file is save and enter the following command: hledger -f demo.hledger stats. If you don't get an error,
+* congratulations, you have done your first transactions correctly!!!
+
+
+
+
+
+
+
+
+; The next transaction are expenses payed using a credit card. Note that the amount doesn't leave you account immediately. You will pay the bank later in the month,
+; so it is an amount you OWE to the bank, in other words, a Liability.
+2018/01/16 Wesley Pub  ; I don't need either the code nor the payee
+    Liabilities:Credit Card:Visa Platinum       -150  ; I could have use only 'visa' as the account name because hledger would recognized as an alias defined above, but I preferred not
+    Expenses:Leisure:Events                           ; Remember Expenses are positive, the opposite of Liabilities
+
+2018/01/17 Calvin Klein
+    Expenses:Products:Clothing                    80
+    Liabilities:Credit Card:Visa Platinum             ; It doesn't matter the posting order nor which of them is implicit
+
+2018/01/20 Visa | Invoice                             ; Paying the credit card invoice
+    Asset:Bank:JP Morgan                         -230 ; Payment decrease my bank balance
+    Liabilities:Credit Card:Visa Platinum             ; This account will go to zero after this payment, meaning I don't owe anything anymore to the bank 
+
+
+* Now we will start to see how hledger can help us to get a picture of our financial life by running some commands. In your terminal, set LEDGER_FILE variable to be this file path.
+* If you are already on this file's folder type: 'LEDGER_FILE=demo.hledger'. Now run the commands below and analyze the results.
+
+* hledger print --explicit
+* hledger register "Bank:"
+* hledger register ":Credit Card"
+* hledger balance  <-- Sum is zero, our accounting is balanced!
+* hledger balancesheetequity && hledger incomestatement. See that the increase in assets is justified by the same amount in our profit as shown by the incomestatement
+* hledger balance "Expenses:" --weekly --tree
+* hledger balance "Expenses:" --weekly --depth 2 --transpose
+* hledger balance "Bank:" --weekly --row-total --tree
+* hledger balance "Bank:" --weekly --row-total --historical --tree
+* hledger balance "Expenses:" --pivot payee
+
```

Comment:
Here is where I plan to comment about hledger-web and hledger-ui. It is after the user sees some transaction and report. Before the transaction, I think it can be confusing. Do you agree?